gql/client.py

```diff
@@ -293,8 +297,12 @@ def __enter__(self):
             self.session = SyncClientSession(client=self)
 
         # Get schema from transport if needed
-        if self.fetch_schema_from_transport and not self.schema:
-            self.session.fetch_schema()
+        try:
+            if self.fetch_schema_from_transport and not self.schema:
+                self.session.fetch_schema()
+        except Exception:
+            await self.transport.close()
```

Comment:
Should be without the `await` of course

Reply:
What a stupid copy and paste error :see_no_evil: sorry.